agent/agent.go

```diff
@@ -478,7 +478,7 @@ func (a *Agent) Start() error {
 	a.serviceManager.Start()
 
 	// Load checks/services/metadata.
-	if err := a.loadServices(c); err != nil {
+	if err := a.loadServices(c, nil); err != nil {
```

Comment:
The reason why we don't need to pass in a snapshot is that the agent just started and it doesn't have any information in memory about anything pretty much.